controllers/openstackdataplanenode_controller.go

```diff
@@ -236,3 +248,139 @@ func (r *OpenStackDataPlaneNodeReconciler) GenerateInventory(ctx context.Context
+	if instance.Spec.Node.NetworkConfig != instanceRole.Spec.NodeTemplate.NetworkConfig {
+		return fmt.Sprintf("{template: %s}", instance.Spec.Node.NetworkConfig)
+
+	}
+	return fmt.Sprintf("{template: %s}", instanceRole.Spec.NodeTemplate.NetworkConfig.Template)
+}
+
+// GetAnsibleNetworks retruns a JSON string mapping fixedIP and/or network name to their valules
+func (r *OpenStackDataPlaneNodeReconciler) GetAnsibleNetworks(instance *dataplanev1beta1.OpenStackDataPlaneNode, instanceRole *dataplanev1beta1.OpenStackDataPlaneRole) string {
+	var network string // the resulting string containing each network
+	for _, netMap := range instance.Spec.Node.Networks {
+		if netMap.FixedIP != "" && netMap.Network == "" {
+			network += fmt.Sprintf("{%s: %s},", "fixedIP", netMap.FixedIP)
+		}
+		if netMap.FixedIP != "" && netMap.Network != "" {
+			network += fmt.Sprintf("{%s: %s, %s: %s},",
+				"fixedIP", netMap.FixedIP, "network", netMap.Network)
+		}
+	}
+	return fmt.Sprintf("[%s]", strings.TrimSuffix(network, ","))
+}
+
+// EnsureNodeOnRoleList will ensure that a node is on its role's dataPlaneNodes list
+func (r *OpenStackDataPlaneNodeReconciler) EnsureNodeOnRoleList(ctx context.Context, instance *dataplanev1beta1.OpenStackDataPlaneNode, instanceRole *dataplanev1beta1.OpenStackDataPlaneRole) error {
+	// 1. When a node is created by a role it will already be on the list
+	// 2. When a node is directly created with a role name, it won't be on the list
+	// In the second case we have to add the node directly and set nodeFrom field
+
+	var err error
+	nodeOnList := false
+	for _, node := range instanceRole.Spec.DataPlaneNodes {
+		if node.HostName == instance.Spec.HostName {
+			nodeOnList = true
+			break
+		}
+		if node.AnsibleHost == instance.Spec.AnsibleHost {
+			nodeOnList = true
+			break
+		}
+	}
+	if !nodeOnList {
+		// TODO(fultonj): append node with name and nodeFrom to instanceRole.Spec.DataPlaneNodes
+
+		nodeSection := &dataplanev1beta1.OpenStackDataPlaneNodeSpec{}
+		nodeSection.NodeFrom = instance.Name
+
+		// Debug type and slice information:
+		// fmt.Println(nodeSection)
+		// fmt.Println(len(instanceRole.Spec.DataPlaneNodes))
+		// fmt.Println(cap(instanceRole.Spec.DataPlaneNodes))
+		// fmt.Printf("%T", &instanceRole.Spec.DataPlaneNodes[0])
+		// fmt.Println()
+		// fmt.Printf("%T", nodeSection)
+		// fmt.Println()
+		// fmt.Printf("%T", instanceRole.Spec.DataPlaneNodes)
+		// fmt.Println()
+
+		// I plan to wrap this in a CreateOrUpdate(), but I can't
+		// seem to define what will be in the mutate function.
+		// This works:
+		//
+		//   instanceRole.Spec.DataPlaneNodes[0] = *nodeSection
+		//
+		// But I want to append to len()+1
+		// But that fails since cap == len
+		// I know append() will increase the slice's cap, e.g.
+		//
+		//   instanceRole.Spec.DataPlaneNodes.append(*nodeSection)
```

Comment:
Looks like `OpenStackDataPlaneNodeSpec` needs an append method.

Reply:
not sure it works:
```go
instanceRole.Spec.DataPlaneNodes = append(instanceRole.Spec.DataPlaneNodes, nodeSection)
```

Reply:
@fao89 I tried that but I got the following:
```
# github.com/openstack-k8s-operators/dataplane-operator/controllers
vet: controllers/openstackdataplanenode_controller.go:368:79: cannot use nodeSection (variable of type *"github.com/openstack-k8s-operators/dataplane-operator/api/v1beta1".OpenStackDataPlaneNodeSpec) as "github.com/openstack-k8s-operators/dataplane-operator/api/v1beta1".OpenStackDataPlaneNodeSpec value in argument to append
```
I'm working on updating one of the following to define a function to do the append on the list.

```
	modified:   api/v1beta1/openstackdataplanenode_types.go
	modified:   api/v1beta1/openstackdataplanerole_types.go
```

Reply:
This worked:
`instanceRole.Spec.DataPlaneNodes = append(instanceRole.Spec.DataPlaneNodes, *nodeSection)`

Thank you @fao89 !